spring-cloud-sleuth-zipkin-stream/src/main/java/org/springframework/cloud/sleuth/zipkin/stream/ConvertToZipkinSpanList.java

```diff
@@ -146,4 +146,21 @@ private static boolean hasClientSend(Span span) {
 		return false;
 	}
 
+	private static long calculateDurationInMicros(Span span) {
```

Comment:
I think I see what's going on here.. the goal being to not confuse by having a span wider than cr - cs. Seems a shame that this means we can't use the precise duration... probably warrants a comment or javadoc explaining this. iirc, it would be something like..

There could be instrumentation delay between span creation and the semantic start of the span (client send). When there's a difference, spans look confusing. Ex users expect duration to be client receive - send, but it is a little more than that. Rather than have to teach each user about the possibility of instrumentation overhead, we truncate absolute duration (span finish - create) to semantic duration (client receive - send)